src/wrappers/foresight/TestRunnerSupport.ts

```diff
@@ -0,0 +1,179 @@
+
+const findHostName = () => {
+    return os.hostname();
+};
+
+const getTestRunId = () => {
+    let testRunId: string;
+    const ei: EnvironmentInfo = EnvironmentSupport.getEnvironmentInfo();
+    if (ei != null) {
+        testRunId = ei.testRunId;
+    } else {
+        testRunId = Utils.generateId();
+    }
+
+    return testRunId;
+};
+
+const hostName: string = findHostName();
+let projectId: string;
+let testStatusReportFreq: number;
+let testRunStatusWork: NodeJS.Timeout;
+
+export let testSuiteName: string;
+export let wrapperContext: WrapperContext;
+export let initialized: boolean = false;
+export let testRunScope: TestRunScope;
+export let testSuiteExecutionContext: TestSuiteExecutionContext;
+export let testCaseExecutionContext: TestCaseExecutionContext;
+
+export const setProjectId = (id: string) => {
+    projectId = id;
+};
+
+export const setTestSuiteName = (name: string) => {
+    testSuiteName = name;
+};
+
+export const setWrapperContext = (context: WrapperContext) => {
+    wrapperContext = context;
+};
+
+export const setInitialized = (value: boolean) => {
+    initialized = value;
+};
+
+export const setTestSuiteContext = (context: ExecutionContext) => {
+    testSuiteExecutionContext = context as TestSuiteExecutionContext;
+};
+
+export const setTestCaseContext = (context: ExecutionContext) => {
+    testCaseExecutionContext = context as TestCaseExecutionContext;
+};
+
+export const setTestStatusReportFreq = (freq: number) => {
+    testStatusReportFreq = freq;
+};
+
+/** if needed keep suite and cases contexts in here
+ * export const testSuiteContextMap = new Map<string, TestSuiteExecutionContext>();
+ * export const testCaseScopeMap = new Map<string, TestCaseExecutionContext>();
+ */
+
+const sendTestRunStatus = async () => {
+    if (!testRunScope || !initialized) {
+
+        ThundraLogger.debug('<ThundraRunnerSupport> Test run not started. TestRunStatus will not send.');
+        return;
+    }
+
+    const config = ConfigProvider.thundraConfig;
+    const { apiKey } = config;
+
+    const statusTimestamp = new Date().getTime();
+    const testRunStatus = TestRunStatus
+        .builder()
+            .withId(testRunScope.id)
+            .withProjectId(projectId)
+            .withTaskId(testRunScope.taskId)
+            .withStartTimestamp(testRunScope.startTimestamp)
+            .withStatusTimestamp(statusTimestamp)
+            .withTotalCount(testRunScope.context.totalCount)
+            .withSuccessfulCount(testRunScope.context.successfulCount)
+            .withFailedCount(testRunScope.context.failedCount)
+            .withIgnoredCount(testRunScope.context.ignoredCount)
+            .withAbortedCount(testRunScope.context.abortedCount)
+            .withHostName(hostName)
+            .withEnvironmentInfo(EnvironmentSupport.getEnvironmentInfo())
+        .build();
+
+    await wrapperContext.reporter.sendReports([Utils.generateReport(testRunStatus, apiKey)]);
+    startTestRunStatusEvent();
```

Comment:
If reporting is failed somehow, sending testrun-status events is not scheduled here. So it might be better to `startTestRunStatusEvent()` into finally block:
```js
try {
    await wrapperContext.reporter.sendReports([Utils.generateReport(testRunStatus, apiKey)]);
} finally {
    startTestRunStatusEvent();
}
```
- Schedule `startTestRunStatusEvent` with